tracer/src/Datadog.Trace/RemoteConfigurationManagement/RemoteConfigurationManager.cs

```diff
@@ -72,13 +78,36 @@ internal class RemoteConfigurationManager : IRemoteConfigurationManager
         {
             lock (LockObject)
             {
-                return Instance ??= new RemoteConfigurationManager(
+                Instance ??= new RemoteConfigurationManager(
                     discoveryService,
                     remoteConfigurationApi,
                     id: settings.Id,
                     rcmTracer: new RcmClientTracer(settings.RuntimeId, settings.TracerVersion, serviceName, environment, serviceVersion),
                     pollInterval: settings.PollInterval);
             }
+
+            while (_initializationQueue.TryDequeue(out var action))
+            {
+                action(Instance);
+            }
+
+            return Instance;
+        }
+
+        public static void GetInitializedInstance(Action<RemoteConfigurationManager> action)
```

Comment:
The naming gives the impression it's a function but it's void and not returning the Instance, maybe it couldbe something like `OnInitialized`?

Reply:
fixed in ad3f7961e